cfgrammar/src/lib/span.rs

```diff
@@ -152,3 +152,37 @@ impl NewlineToLineColCache {
         })
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::NewlineToLineColCache;
+    use std::ops::Range;
+    #[test]
+    fn newline_cache_splits() {
+        let same_line_col_1 = [
```

Comment:
I'll have to digest this in the morning, sorry for adding more copy/pastes of the above test you quote, but I wanted to fix/check that case I just added so I could sleep :rofl:

Reply:
I've added `newline_test_helper`, I went with the `&[Vec<(usize, usize)>]`, but I didn't have a strong preference.
It shows the difference between these `["a", "b"]` cases, but maybe flattening these is preferred, we wouldn't have to repeat so much, but it wouldn't show the difference anymore :shrug:.

I also called `feed` with all the arguments first, because I think it will catch it would catch cases where we could `trailing_bytes` when it is out of date because another string has been fed, rather than doing it in the single loop.

Anyhow just added a couple of cases, until we've decided